src/sql/parts/notebook/models/cell.ts

```diff
@@ -458,4 +465,16 @@ export class CellModel implements ICellModel {
 		// Use this to set the internal (immutable) and public (shared with extension) uri properties
 		this.cellUri = uri;
 	}
+
+	// Get Knox endpoint from IConnectionProfile
+	// TODO: this will be refactored out into the notebooks extension as a contribution point
+	private getKnoxEndpoint(activeConnection: IConnectionProfile): notebookUtils.IEndpoint {
+		let endpoint;
+		if (this._connectionManagementService) {
```

Comment:
providerId === MSSQL only, if activeConnection is empty don't do it, and also check if 'clusterEndpoints' is non-null since it's only existing on SQL big data cluster endpoints.